### PR TITLE
Dovecot lucene pull req

### PR DESCRIPTION
**Pull request by brocktice**
This sets everything up for using the official MIAB PPA to serve a dovecot-lucene package compatible with the official dovecot-core. It also makes all changes necessary to enable lucene for full-text search in dovecot.

The source package will have to be built and uploaded to the PPA by the maintainer, after which launchpad will build the package. There is a working version of the package on my PPA here:

https://launchpad.net/~brock/+archive/ubuntu/mailinabox-brocktice

Please accept my apologies for any problems here, I have done my best to correctly cherry-pick the required changes.

### CODE REVIEW

**Comment by JoshData** [comment]
Nice work. I'm going to leave it open for people to test for a while before merging.

(Adding the PPA will also upgrade postgrey to my fork, so the merge is dependent on that working also. See #400.)

**Comment by brocktice** [comment]
Sounds good. People wanting to test can use my fork if they'd like:

https://github.com/brocktice/mailinabox/tree/lucene

I've got it set back to my ppa for now so they don't have to set up their own to test.

File: setup/lucene.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+}
+EOF
+
+# PERMISSIONS
+
+# Ensure configuration files are owned by dovecot and not world readable.
+chown -R mail:dovecot /etc/dovecot
+chmod -R o-rwx /etc/dovecot
+
+# Restart services to reload solr schema & dovecot plugins
```
Comment (Jonty) [inline setup/lucene.sh]: Should probably say "lucene" rather than solr, just so people don't get confused!

Comment (brocktice) [inline setup/lucene.sh]: OK if I fix this and push to the appropriate branch it should add to the PR right?

**Comment by JoshData** [comment]
Thanks for the work on this! I made some simplifications and merged it in.

**Pull request merged**